src/mesg.c

```diff
@@ -262,8 +290,10 @@ send_nick(char *err, char *mesg)
 
 	if ((nick = strtok(mesg, " ")))
 		return sendf(err, ccur->server, "NICK %s", mesg);
+        else
+                fail("Error: NICK requires a nick");
```

Comment:
This makes line 266 unreachable because you either return sendf or fail. The goal with /nick was to have it print the user's current nick for this server if no argument is present. The reason for the segfault is that the initial buffer at startup has no server. Maybe check for c->server and print config.nicks if NULL?